spyder/app/mainwindow.py

```diff
@@ -34,6 +34,8 @@
 import threading
 import traceback
 
+from attr import has

```

Comment:
- Please move this import to be before `from collections import OrderedDict`-
- Also remove `from __future__ import print_function` because it's no longer necessary.